### PR TITLE
common.library: include helper to handle dns-valid names with suffixes

### PR DESCRIPTION
**Pull request by sigilioso**
#### Special notes for your reviewer:

Include a naming helper to handle names with prefixes. See potential usage example [here](https://github.com/newrelic/k8s-metadata-injection/blob/bb45f8a0c1d49fd47fa96bb95cd14b8603ce327f/charts/nri-metadata-injection/templates/admission-webhooks/job-patch/job-createSecret.yaml#L5)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [ ] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### CODE REVIEW

File: library/common-library/templates/_naming.tpl [hunk 1]
```diff
@@ -8,6 +8,27 @@ This is an function to be called directly with a string just to truncate strings
...
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+It fails if suffix is too long to fint in the limit set.
+Usage:
+{{ include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" "<my-name>" "suffix" "my-suffix" ) }}
+*/ -}}
+{{- define "newrelic.common.naming.truncateToDNSWithSuffix" -}}
+{{- $maxLen := (sub 63 (len .suffix)) -}}
+
+{{- if (lt $maxLen 1) -}}
+    {{ fail "(newrelic.common.naming.truncateToDNSWithSuffix) suffix length is too long to compose a valid name" }}
```
Comment (kang-makes) [inline library/common-library/templates/_naming.tpl:22]: NIT: This error is not specific enough and is going to be shown to customers... It should be more specific.
```suggestion
    {{ fail (printf "The release name you choose (%s) is hitting the Kubernetes limit of 63 chars for the object name. We cannot create %s-%s" .name .name .suffix) }}
```
I did not test my suggestion but you should get the idea ;)

**Review by kang-makes — APPROVED** [review]
Sounds cool. LGTM but I would rephrase the error of that helper.

File: library/common-library/templates/_naming.tpl (hunk 1 above)
Comment (sigilioso) [reply library/common-library/templates/_naming.tpl:22]: The problem would not be the release name but the suffix chosen instead, anyway I like the idea of being more specific, thanks!

Comment (sigilioso) [reply library/common-library/templates/_naming.tpl]: Error improved here: [b4ecebf](https://github.com/newrelic/helm-charts/pull/785/commits/b4ecebf7ba33d0cdefd6f0d8708d1d4413248759)

File: library/common-library/templates/_naming.tpl [hunk 2]
```diff
@@ -8,6 +8,27 @@ This is an function to be called directly with a string just to truncate strings
...
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+It fails if suffix is too long to fint in the limit set.
+Usage:
+{{ include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" "<my-name>" "suffix" "my-suffix" ) }}
+*/ -}}
+{{- define "newrelic.common.naming.truncateToDNSWithSuffix" -}}
+{{- $maxLen := (sub 63 (len .suffix)) -}}
+
+{{- if (lt $maxLen 1) -}}
+    {{ fail (printf "The suffix chosen (%s) is hitting the Kubernetes limit of 63 chars for the object name. We cannot create %s-%s" .suffix .name .suffix) }}
```
Comment (gsanchezgavier) [inline library/common-library/templates/_naming.tpl:22]: would make sense to warn instead of fail? not sure if there are any implications on this being chopped more than "cosmetic"

Comment (sigilioso) [reply library/common-library/templates/_naming.tpl:22]: I think there could be unexpected name collisions (it actually depends on how the helper is used). Normally it would be cosmetic but we cannot be sure...

Comment (sigilioso) [reply library/common-library/templates/_naming.tpl]: As discussed, failing here is not coherent with truncating long names when there is no suffix, so I changed the helper to truncate not only the name but also the suffix before using it: [94867bc](https://github.com/newrelic/helm-charts/pull/785/commits/94867bcfb9e75d11e708c30a1836001a2d6cb47a)

File: library/common-library/templates/_naming.tpl [hunk 3]
```diff
@@ -11,19 +11,20 @@ This is an function to be called directly with a string just to truncate strings
 {{- /*
 Given a name and a suffix returns a 'DNS Valid' which always include the suffix, truncating the name if needed.
 We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
-It fails if suffix is too long to fint in the limit set.
+If suffix is too long it gets truncated but it al`ays takes precedence over name, so a 63 chars suffix would suppress the name.
```
Comment (gsanchezgavier) [inline library/common-library/templates/_naming.tpl:14]: typo: always

Comment (sigilioso) [reply library/common-library/templates/_naming.tpl]: thanks, good catch! 👍
